src/Serialization/upgrades/0.11.2.jl

```diff
@@ -0,0 +1,91 @@
+################################################################################
+# Upgrade Summary
+# Types of the form `Vector{T}` are only serialized with backref when
+# `is_basic_serialization_type(T) == false`. If
+# `is_basic_serialization_type(T) == true` the type `Vector{T}` does not
+# serialize `T` in the entries, but instead as a part of the `Vector` serialization.
+# Instances of the form `Tuple{T}` with `is_basic_serialization_type(T) == true` 
+# are no longer serialized with backref, however these types were already storing
+# there types outside of the JSON array of entries
+
+push!(upgrade_scripts, UpgradeScript(
+    v"0.11.2",
+    function (s::DeserializerState, dict::Dict) 
+        if haskey(dict, :_ns)
+            haskey(dict[:_ns], :polymake) && return dict
+        end
+    
+        if !haskey(dict, :type)
+            upgraded_dict = Dict{Symbol, Any}()
+            for (key, value) in dict
+                if value isa String
+                    upgraded_dict[key] = value
+                else
+                    # recursive call
+                    upgraded_dict[key] = var"#self#"(s, value)
+                end
+            end
+            return upgraded_dict
+        end
+
+        !haskey(dict, :data) && return dict
+        
+        if dict[:type] == string(backref_sym)
+            backref = s.objs[UUID(dict[:id])]
+            backref_type = decodeType(backref[:type])
+
+            # recursive call
+            backref_type <: OscarBasicType && return var"#self#"(s, backref)
+            return dict
+        end
+
+        if dict[:type] == "Vector"
```

Comment:
I feel this upgrade function badly needs comments. What is going on here? Right now I think I could cobble it together from looking at this PR, but if I would have to debug this code in half a year, I'd be lost. Can you perhaps at least add a comment before each of the main `if` that states what it is about?